tox.ini

```diff
@@ -0,0 +1,61 @@
+[tox]
+envlist =
+    py{36,37,38,39}-test{,-alldeps,-oldestdeps}{,-devastropy}{,-cov}
+    codestyle
+    build_docs
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+    tox-pypi-filter >= 0.12
+isolated_build = true
+
+
+[testenv]
+
+# The following option combined with the use of the tox-pypi-filter above allows
+# project-wide pinning of dependencies, e.g. if new versions of pytest do not
+# work correctly with pytest-astropy plugins. Most of the time the pinnings file
+# should be empty.
+pypi_filter = https://raw.githubusercontent.com/astropy/ci-helpers/master/pip_pinnings.txt
+
+# Run the tests in a temporary directory to make sure that we don't import
+# astropy from the source tree
+changedir = .tmp/{envname}
+
+description = run tests
+
+deps =
+    devastropy: git+https://github.com/astropy/astropy.git#egg=astropy
+
+# TODO: Add more versions to oldestdeps
+    oldestdeps: astropy==3.1
+
+extras =
+    test
+    alldeps: all
+
+commands =
+    pip freeze
+# FIXME: there are too many failures from the docs example gallery ignore docs for now
+# !cov: pytest {toxinidir}/astroquery {toxinidir}/docs
+    !cov: pytest {toxinidir}/astroquery
+    cov:  pytest {toxinidir}/astroquery --cov {toxinidir}/astroquery
+    cov: coverage xml -o {toxinidir}/coverage.xml
+
+[testenv:codestyle]
+changedir = {toxinidir}
+skip_install = true
+description = check code style
+deps = pycodestyle
+commands = pycodestyle astroquery --count
+
+
+[testenv:build_docs]
+changedir =
+description = Building the narrative and API docs
+extras = docs
+requires =
+    sphinx < 3.4
```

Comment:
I am not sure if this incantation works or why you pinning to `<3.4`. I see this in the log:
```
Running Sphinx v3.4.2
```

Reply:
I don't need to, it's a remnant of some local try and error